e2e: Fix shadowed error in reboot test and clean up containers after reboot test

rzlink opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

#### What type of PR is this?
/kind bug
/kind flake
/sig windows

<!--
Add one of the following kinds:
/kind bug
/kind cleanup
/kind documentation
/kind feature

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind deprecation
/kind failing-test
/kind flake
/kind regression
-->

#### What this PR does / why we need it:
- Fixed a bug in function `ExecCommandInContainerWithFullOutput` usage in `hybrid_network.go` where the `err` variable was shadowed, causing test failures to be silently ignored.

- Added cleanup logic in `eviction.go` to explicitly delete pods created by the reboot node test (e.g., `img-puller`, `reboot-host-test-windows`) before starting the eviction test.

- This improves reliability of the eviction test when the reboot test runs beforehand and leaves behind memory-consuming pods.

#### Which issue(s) this PR fixes:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #(https://github.com/kubernetes/kubernetes/issues/130419)

#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

rzlink committed:
Fix test flakiness caused by shadowed error in reboot test and stale pods affecting eviction test

- Fixed a bug in `ExecCommandInContainerWithFullOutput` usage in `hybrid_network.go` where the `err` variable was shadowed, causing test failures to be silently ignored.
- Added cleanup logic in `eviction.go` to explicitly delete pods created by the reboot node test (e.g., `img-puller`, `reboot-host-test-windows`) before starting the eviction test.
- This improves reliability of the eviction test when the reboot test runs beforehand and leaves behind memory-consuming pods.

rzlink commented:
/assign @marosset @jsturtevant

marosset commented:
/ok-to-test

marosset commented:
/test pull-kubernetes-e2e-capz-windows-master-serial-slow

rzlink committed:
Add cleanup containers in reboot node test

zylxjtu commented on test/e2e/windows/reboot_node.go:
```diff
@@ -269,3 +276,104 @@ var _ = sigDescribe(feature.Windows, "[Excluded:WindowsDocker] [MinimumKubeletVe
+	user := "NT AUTHORITY\\SYSTEM"
+	windowsImage := imageutils.GetE2EImage(imageutils.Agnhost)
+
+	cmd := `$lines = & "$Env:ProgramFiles\\containerd\\ctr.exe" -n k8s.io containers list | Select-String e2e-test-images;
```
is "e2e-test-images" the namespace name?

rzlink replied:
No, `e2e-test-images` is part of the image URL. The `cri.exe containers list` command only shows the container ID and image—it doesn't include the namespace. To get the namespace, you need to run `cri.exe containers info`.

zylxjtu replied:
Can we use the image name instead? I'm thinking that if we use a private registery, it may not have the e2e-test-images in the URL

zylxjtu commented on test/e2e/windows/reboot_node.go:
```diff
@@ -269,3 +276,104 @@ var _ = sigDescribe(feature.Windows, "[Excluded:WindowsDocker] [MinimumKubeletVe
+		}
+	}
+	Write-Host "Rebooting host";
+	shutdown.exe -r -t 5;`
```
do we need another reboot?

rzlink replied:
This is part of the cleanup phase—one reboot is sufficient here.

marosset commented:
/test pull-kubernetes-e2e-capz-windows-master-serial-slow

marosset commented:
/test pull-kubernetes-e2e-capz-windows-serial-slow

rzlink commented:
/test pull-kubernetes-e2e-capz-windows-serial-slow

rzlink committed:
Workaround for Calico HNS issue by avoiding eviction of undeletable pods

Add a workaround for a known Calico issue (https://github.com/projectcalico/calico/issues/6974) where pods on Windows nodes may become undeletable after a reboot. This causes the eviction manager to attempt evicting these pods but fail due to HNS namespace deletion errors.

The workaround avoids scheduling critical test pods on rebooted nodes to prevent interference.

TODO: Remove this workaround once the Calico issue is resolved.

rzlink committed:
Fix the lint issue.

rzlink commented:
/test pull-kubernetes-e2e-capz-windows-master-serial-slow

rzlink commented:
/test pull-kubernetes-e2e-capz-windows-serial-slow

rzlink commented:
/test pull-kubernetes-e2e-capz-windows-serial-slow

marosset commented:
/label tide/merge-method-squash

marosset commented on test/e2e/windows/kubelet_stats.go:
```diff
@@ -234,6 +234,29 @@ func findWindowsNode(ctx context.Context, f *framework.Framework) (v1.Node, erro
+	}
+
+	if len(targetNodes) == 0 {
+		e2eskipper.Skipf("Could not find and ready and schedulable Windows nodes")
```
I don't think we want to skip in this method, and if we do please indicate that will happen in the name and function comment.
I think it would be better to return the list of nodes and let each test case decide if it should skip or not.

rzlink committed:
Address comments

rzlink commented:
/test pull-kubernetes-e2e-capz-windows-serial-slow

marosset commented:
/test pull-kubernetes-e2e-capz-windows-serial-slow

rzlink commented:
/test pull-kubernetes-e2e-capz-windows-serial-slow

rzlink commented:
/test pull-kubernetes-e2e-capz-windows-serial-slow

marosset reviewed (approved):
/approve

zylxjtu commented:
/lgtm

rzlink commented:
/retest-required
